### PR TITLE
feat: `defineEnv` util

### PR DESCRIPTION
**Pull request by pi0**
This PR adds a new `defineEnv({ presets: [], ... }): { env }` utility aiming to replace `env(presets[]): env` util for integration usage.

The benefit of this one is that it can be extendable with more features, options and shortcuts.

Example:

```ts
const { env } = defineEnv({
 nodeCompat: true,
 presets: [myPreset],
 overrides: {}
});
```

### CODE REVIEW

**Comment by pi0** [comment]
(merging as experimental to iterate and improve API features)

File: src/env.ts [hunk 1]
```diff
@@ -1,5 +1,44 @@
...
+    presets.push(...opts.presets);
+  }
+
+  if (opts.overrides) {
+    presets.push(opts.overrides);
+  }
+
+  const resolvedEnv = env(...presets);
+
+  return { env: resolvedEnv };
```
[the remark this answers is not in the document]
Comment (pi0) [reply src/env.ts:35]: Both for extendability (you see in next PR!)

File: src/types.ts [hunk 2]
```diff
@@ -1,3 +1,20 @@
+export interface CreateEnvOptions {
+  /**
+   * Enable Node.js compatibility (nodeless) preset.
```
Comment (vicb) [inline src/types.ts:3]: Explicitly add the default value

Comment (pi0) [reply src/types.ts:3]: I think **Enable ...** implies that default is false won't it?

Comment (pi0) [reply src/types.ts:3]: https://github.com/unjs/unenv/pull/373/files#diff-c54113cf61ec99691748a3890bfbeb00e10efb3f0a76f03a0fd9ec49072e410aR5

File: src/env.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pi0) [reply src/env.ts:35]: - #373 updates resolvedEnv before return
- #374 exposes presets alongside env

**Pull request merged**